src/project/internal/notationproject.cpp

```diff
@@ -641,10 +653,26 @@ Ret NotationProject::doSave(const muse::io::path_t& path, engraving::MscIoMode i
                 LOGW() << ret.toString();
             }
         } else {
-            Ret ret = fileSystem()->move(savePath, targetContainerPath, true);
+            Ret ret = checkSavedFileForCorruption(ioMode, savePath, targetMainFileName.toQString());
+            if (!ret) {
+                if (ret.code() == (int)Err::CorruptionUponSavingError) {
+                    ret.setData("corruptedAfterMove", false);
+                }
+                return ret;
+            }
+
+            ret = fileSystem()->move(savePath, targetContainerPath, true);
             if (!ret) {
                 return ret;
             }
+
+            ret = checkSavedFileForCorruption(ioMode, targetContainerPath, targetMainFileName.toQString());
```

Comment:
I originally thought we would only check it once, but we're checking it 2 times, which is overkill. Instead, I would suggest keeping only this check, so that we won't increase the saving time even further